docs: Add query README file and explain some rationale (#648)
influxdata/influxdb_iox: query/README.md

# IOx Query Layer

The IOx query layer is responsible for translating query requests from
different query languages and planning and executing them against
Chunks stored across various IOx storage systems.

Query Frontends
* SQL
* Storage gRPC
* Flux (possibly in the future)
* InfluxQL (possibly in the future)
* Others (possibly in the future)

Sources of Chunk data
* ReadBuffer
* MutableBuffer
* Parquet Files

> * Others (possibly in the fugure, like Remote Chunk?)

Reviewer: ```suggestion
* Others (possibly in the future, like Remote Chunk?)
```

The goal is to use the shared query / plan representation in order to

> avoid N*M combinations of languge and Chunk source.

Reviewer: ```suggestion
avoid N*M combinations of language and Chunk source.
```

Thus query planning is implemented in terms of traits, and those

> traits are implemented by different chunk impementations.

Reviewer: ```suggestion
traits are implemented by different chunk implementations.
```

> Amont othe rthings, this means that this crate should not depend

Reviewer: ```suggestion
Among other things, this means that this crate should not depend
```
